src/map/unit.cpp

```diff
@@ -246,6 +286,11 @@ int32 unit_walktoxy_sub(struct block_list *bl)
 		unit_refresh( bl, true );
 	}
 #endif
+
+	// Set mobstate here already as chase skills can be used on the first frame of movement
+	// If we don't set it now the monster will always move a full cell before checking
+	set_mobstate(bl, ud->state.attack_continue);
```

Comment:
Since we more or less have a half switch for the different bl_types above already.
Could you change it to a switch (bl->type) please and merge the code into it?